### DISCUSSION TITLE
Feature Request: Method to suppress output from `CREATE SECRET`

### DISCUSSION THREAD

**Discussion by daviewales** [duckdb/duckdb#12821] [Ideas]
Why do you want this feature?

Suppose I want to run a query and output comma separated table data for a table in Azure blob storage:

```
duckdb -csv-c "load azure; SET azure_transport_option_type = 'curl'; CREATE SECRET example (TYPE AZURE, PROVIDER CREDENTIAL_CHAIN, ACCOUNT_NAME 'example'); describe from 'az://example/data/table.parquet'"
```

What I want is just the data from the query:

```
column_name,column_type,null,key,default,extra
column1,VARCHAR,YES,,,
column2,VARCHAR,YES,,,
```

What I get is extra data from `CREATE SECRET`:

```
Success
true
column_name,column_type,null,key,default,extra
column1,VARCHAR,YES,,,
column2,VARCHAR,YES,,,
```

Of course, I can work around this by piping the output through `tail -n +3`.
But if would be nice if there was a way to suppress the output of `CREATE SECRET` directly in duckdb.

**Comment by Tishj** [comment]
I assume this is a request for the CLI?
I think there is a similar option for create table/inserts/updates that is off by default in the CLI

**Comment by daviewales** [comment]
Yes, the use case is I was trying to use the output in a script, but I had strip out the output from CREATE SECRET. I was thinking it would be cleaner if I could set an option so only the output from actual data queries was returned.

**Reply by Tishj** [reply]
For changed rows there is a `.changes` setting:
`.changes on|off          Show number of rows changed by SQL`

I guess it makes sense to have this function similar to that

**Reply by Tishj** [reply]
Actually.. I don't think that's the right approach.
I can't immediately see a good reason why this is returning a query result instead of CHANGED_ROWS, in which case it will be filtered out by the `changes` setting as well.

**Reply by Tishj** [reply]
@samansmink is there a specific reason this uses `StatementReturnType::QUERY_RESULT` instead of `StatementReturnType::CHANGED_ROWS`, returning 1 instead?

Or even `StatementReturnType::NOTHING`, an error would be thrown instead if it wasn't successful so the `true` is a bit redundant maybe?
`NOTHING` is the default for other CREATE statements

**Reply by samansmink** [reply]
hmm @Tishj I actually have a [PR](https://github.com/duckdb/duckdb/pull/12504) open that will add some extra information to the secret creation result. The idea being that especially for secrets that use credential chains it is nice to print the result of the creation  to the screen right away: this way a users is informed right away which credentials were used for secret creation. 

However I think it indeed it makes more sense to just create it and let the user query `duckdb_secrets()` to figure out which credentials were used and have create secret use `StatementReturnType::NOTHING`.

**Comment by nicornk** [comment]
+1 to this. I am using "-cmd" to pre-create Views and I always have 

┌─────────┐
│ Success │
│ boolean │
├─────────┤
│ true    │
└─────────┘

in the output.
